Description: Fix Segmentation fault in ShowCache::handleCacheEvent
Author: zhang <15535135608@163.com>
Origin: upstream
Bug: https://github.com/apache/trafficserver/issues/4328
Applied-Upstream: https://github.com/apache/trafficserver/commit/616eb10bfc35599a2c93ff30879d584a05ddf83e
Reviewed-by: Jean Baptiste Favre <debian@jbfavre.org>
Last-Update: 2018-10-17
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/iocore/cache/CachePages.cc
+++ b/iocore/cache/CachePages.cc
@@ -337,6 +337,13 @@ ShowCache::handleCacheEvent(int event, E
     CacheHTTPInfoVector *vec = &(cache_vc->vector);
     int alt_count            = vec->count();
     if (alt_count) {
+      // check cache_vc->first_buf is NULL, response cache lookup busy.
+      if (cache_vc->first_buf == nullptr) {
+        cache_vc->do_io_close(-1);
+        CHECK_SHOW(show("<H3>Cache Lookup Busy, please try again</H3>\n"));
+        return complete(event, e);
+      }
+
       Doc *d = (Doc *)(cache_vc->first_buf->data());
       time_t t;
       char tmpstr[4096];
